core/base/document.php

```diff
@@ -199,10 +199,21 @@ public static function get_class_full_name() {
 		return get_called_class();
 	}
 
+	// TODO: Backward compatibility - Replace with `get_new_create_url`.
 	public static function get_create_url() {
-		$base_create_url = Plugin::$instance->documents->get_create_new_post_url( Source_Local::CPT );
+		Plugin::$instance->modules_manager->get_modules( 'dev-tools' )->deprecation->deprecated_function( __FUNCTION__, '3.3.0', 'get_new_create_url' );
 
-		return add_query_arg( [ 'template_type' => static::get_type() ], $base_create_url );
+		return self::get_new_create_url( Source_Local::CPT, static::get_type() );
+	}
+
+	public static function get_new_create_url( $post_type = Source_Local::CPT, $template_type = null ) {
```

Comment:
Remove and keep the old one. It shouldn't receive args. it should create a URL with `static::get_type()` and `$properties['cpt']` if exist.